### DISCUSSION TITLE
Installer generates a lot of tmp files upon invoking custom actions and doesn't clean them up

### DISCUSSION THREAD

**Discussion by perevodchyk** [wixtoolset/issues#7312] [Questions]
Environment:

- Visual Studio Professional 2022 17.4.1
- WiX Toolset v3.11.2.4516

How to reproduce:

- Run installer built using the WiX Toolset

Problem:

- Every time before a custom action is invoked, the installer generates a temporary file in C:\Windows\Installer which has exactly the same size as InstallerCustomActions.dll. In my case, the file is 44 MB large, and there are 18 files generated during the repair process, which leads to almost 800 MB of redundant data.

**Comment by robmen** [comment]
Wow. That is a very large custom action DLL. It could be the files are scheduled by the Windows Installer for removal after a restart. Are the files still there after you restart the computer?

**Reply by perevodchyk** [reply]
Yes, they remain in the directory even after a restart.
